sonar-pmd-plugin/src/main/java/org/sonar/plugins/pmd/PmdTemplate.java

```diff
@@ -95,7 +96,7 @@ PMDConfiguration configuration() {
     }
 
     public void process(InputFile file, RuleSets rulesets, RuleContext ruleContext) {
-        ruleContext.setSourceCodeFilename(file.uri().toString());
+        ruleContext.setSourceCodeFile(Paths.get(file.uri()).toFile());
```

Comment:
because of PMD master use `RuleContext.sourceCodeFile` to set JavaRuleViolation.filename
```java
 File file = ctx.getSourceCodeFile();
        if (file != null) {
            filename = file.getPath();
        } else {
            filename = "";
        }
```
https://github.com/pmd/pmd/blame/master/pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/ParametricRuleViolation.java#L43